presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/VarcharDecoder.java

```diff
@@ -42,7 +42,7 @@ public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
         if (value == null) {
             output.appendNull();
         }
-        else if (value instanceof String) {
+        else if (value instanceof String || value instanceof Number) {
```

Comment:
The error message in the `else` case needs to be adjusted to say "Expected a string or numeric value for ..."